android/src/main/java/com/apparence/camerawesome/CamerawesomePlugin.java

```diff
@@ -459,6 +475,45 @@ private void _handleAutoFocus(final MethodCall call, final Result result) {
         }
     }
 
+    private void _handleRecordVideo(final MethodCall call, final Result result) {
+        if (!call.hasArgument("path")) {
+            result.error("PATH_NOT_SET", "a file path must be set", "");
+            return;
+        }
+        String path = call.argument("path");
+        try {
+            mCameraPicture.recordVideo(
+                    mCameraStateManager.getCameraDevice(),
+                    path,
+                    mCameraSetup.getOrientation()
+            );
+
+            result.success(null);
+        } catch (CameraAccessException | IOException e) {
+            result.error(e.getMessage(), "cannot open camera", "");
+        }
+    }
+
+    private void _handleStopRecordingVideo(final MethodCall call, final Result result) {
+        mCameraPicture.stopRecording();
```

Comment:
This can crash if the recorder for some reason would be null, should we try-catch and handle it with an error result?

Reply:
I'll add a try catch.